mlflow/gateway/providers/togetherai.py

```diff
@@ -0,0 +1,470 @@
+            #
+            #{'id': '86d8d6e06df86f61-ATH', 'object': 'completion.chunk', 'created': 1711977238, 'choices': [{'index': 0, 'text': '   ', 'logprobs': None, 'finish_reason': None, 'delta': {'token_id': 2287, 'content': '   '}}], 'model': 'mistralai/Mixtral-8x7B-v0.1', 'usage': None}
+            #   {'id': '86d8d6e06df86f61-ATH', 'object': 'completion.chunk', 'created': 1711977238, 'choices': [{'index': 0, 'text': ' "', 'logprobs': None, 'finish_reason': None, 'delta': {'token_id': 345, 'content': ' "'}}], 'model': 'mistralai/Mixtral-8x7B-v0.1', 'usage': None}
+            # "{'id': '86d8d6e06df86f61-ATH', 'object': 'completion.chunk', 'created': 1711977238, 'choices': [{'index': 0, 'text': 'name', 'logprobs': None, 'finish_reason': None, 'delta': {'token_id': 861, 'content': 'name'}}], 'model': 'mistralai/Mixtral-8x7B-v0.1', 'usage': None}
+
+            ## LAST CHUNK
+            #name{'id': '86d8d6e06df86f61-ATH', 'object': 'completion.chunk', 'created': 1711977238, 'choices': [{'index': 0, 'text': '":', 'logprobs': None, 'finish_reason': 'length', 'delta': {'token_id': 1264, 'content': '":'}}], 'model': 'mistralai/Mixtral-8x7B-v0.1', 'usage': {'prompt_tokens': 17, 'completion_tokens': 200, 'total_tokens': 217}}
+            #":[DONE]
+
+        return completions.StreamResponsePayload(
+            id=resp.get("id"),
+            created=resp.get("created"),
+            model=config.model.name,
+            choices=[
+                completions.StreamChoice(
+                    index=idx,
+                    #TODO this is questionable since the finish reason comes from togetherai api
+                    finish_reason=choice.get("finish_reason"),
+                    delta=completions.StreamDelta(
+                        role=None,
+                        content=choice.get("text")
+                    )
+                )
+
+                for idx, choice in enumerate(resp.get("choices", []))
+            ],
+            # usage is not included in OpenAI StreamResponsePayload
+            #usage=completions.CompletionsUsage(
+            #    prompt_tokens=usage.get("prompt_tokens") if usage else None,
+            #    input_tokens=usage.get("completion_tokens") if usage else None,
+            #    total_tokens=usage.get("total_tokens") if usage else None,
+            #)
+        )
+
+    @classmethod
+    def completions_to_model(cls, payload, config):
+        key_mapping = {
+            # repetition_penalty (-3.402823669209385e+38, 3.402823669209385e+38)
+            # frequency_penalty (-2.0, 2.0)
+            # presence_penalty (-2.0, 2.0)
+            "frequency_penalty": "repetition_penalty",
+            # top_logprobs (0, 20)
+            # logprobs (-2147483648,2147483648)
+            "top_logprobs": "logprobs"
+        }
+
+        payload = rename_payload_keys(payload, key_mapping)
+
+        if "logprobs" in payload and payload["logprobs"] and "logprobs" not in payload:
```

Comment:
What is this condition intended to do? It seems the first and last condition contradicts each other.